playwright/tests/index.ts

```diff
@@ -0,0 +1,62 @@
+import { test as base } from '@playwright/test'
+import { mkdtemp, writeFile } from 'node:fs/promises'
+import path from 'path'
+import childProcess from 'child_process'
+import { tmpdir } from 'node:os'
+import serveStatic = require('serve-static')
+import connect = require('connect')
+import * as http from 'node:http'
+
+async function buildApp (projectDirectoryPath: string, bettyConfiguration: object): Promise<string> {
+  // We do not know the real base URL, but as Betty requires one, set an obviously fake value.
+  bettyConfiguration.base_url = 'https://example.com'
+  await writeFile(path.join(projectDirectoryPath, 'betty.json'), JSON.stringify(bettyConfiguration))
+  childProcess.execSync('betty generate', {
+    cwd: projectDirectoryPath
+  })
+  return projectDirectoryPath
+}
+
+class Server implements Disposable {
+  private httpServer: http.Server
+  public readonly publicUrl: string
+
+  constructor (wwwDirectoryPath: string) {
+    const host = '127.0.0.1'
+    const app = connect()
+      .use(serveStatic(wwwDirectoryPath))
+    for (let portNumber = 8000; portNumber <= 65535; portNumber++) {
+      try {
+        this.httpServer = app.listen(portNumber, host)
+        this.publicUrl = `http://${host}:${portNumber}`
+        break
+      } catch (error) {
+        continue
+        if (!error.message.includes('EADDRINUSE')) {
```

Comment:
Local test runs fail on this. I guess that means we do not catch the error correctly after all...

Reply:
See https://nodejs.org/api/net.html#serverlisten